src/Files.App/Services/Windows/WindowsDialogService.cs

```diff
@@ -23,17 +23,7 @@ public unsafe bool Open_FileOpenDialog(nint hWnd, bool pickFoldersOnly, string[]
 			try
 			{
 				using ComPtr<IFileOpenDialog> pDialog = default;
-				var dialogInstanceIid = typeof(FileOpenDialog).GUID;
-				var dialogIid = typeof(IFileOpenDialog).GUID;
-
-				// Get a new instance of the dialog
-				HRESULT hr = PInvoke.CoCreateInstance(
-					&dialogInstanceIid,
-					null,
-					CLSCTX.CLSCTX_INPROC_SERVER,
-					&dialogIid,
-					(void**)pDialog.GetAddressOf())
-				.ThrowOnFailure();
+				HRESULT hr = pDialog.CoCreateInstance(CLSID.CLSID_FileOpenDialog, null, CLSCTX.CLSCTX_INPROC_SERVER);
```

Comment:
```suggestion
				HRESULT hr = pDialog.CoCreateInstance(CLSID.CLSID_FileOpenDialog, null, CLSCTX.CLSCTX_INPROC_SERVER).ThrowOnFailure();
```

I don't think we should drop `.ThrowOnFailure()`.